code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/OptimizeExtensionTests.java

```diff
@@ -2209,6 +2209,260 @@ public void testHandleUpdatePropositionsComplete_missingRequestEventIdInData()
+                new ObjectMapper()
+                        .readValue(
+                                getClass()
+                                        .getClassLoader()
+                                        .getResource("json/EVENT_DATA_EDGE_RESPONSE_VALID.json"),
+                                HashMap.class);
+        final Event testEvent =
+                new Event.Builder(
+                                "AEP Response Event Handle (Spoof)",
+                                "com.adobe.eventType.system",
+                                "com.adobe.eventSource.debug")
+                        .setEventData(edgeResponseData)
+                        .build();
+
+        final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+
+        // test
+        extension.handleDebugEvent(testEvent);
+
+        // verify
+        Mockito.verify(mockExtensionApi, Mockito.times(1)).dispatch(eventCaptor.capture());
+
+        final Event dispatchedEvent = eventCaptor.getValue();
+        Assert.assertEquals("com.adobe.eventType.optimize", dispatchedEvent.getType());
+        Assert.assertEquals("com.adobe.eventSource.notification", dispatchedEvent.getSource());
+
+        final List<Map<String, Object>> propositionsList =
+                (List<Map<String, Object>>) dispatchedEvent.getEventData().get("propositions");
+        Assert.assertNotNull(propositionsList);
+        Assert.assertEquals(1, propositionsList.size());
+
+        final Map<String, Object> propositionsData = propositionsList.get(0);
+        Assert.assertNotNull(propositionsData);
+        final OptimizeProposition optimizeProposition =
+                OptimizeProposition.fromEventData(propositionsData);
+        Assert.assertNotNull(optimizeProposition);
+
+        Assert.assertEquals("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa", optimizeProposition.getId());
+        Assert.assertEquals(
+                "eyJhY3Rpdml0eUlkIjoieGNvcmU6b2ZmZXItYWN0aXZpdHk6MTExMTExMTExMTExMTExMSIsInBsYWNlbWVudElkIjoieGNvcmU6b2ZmZXItcGxhY2VtZW50OjExMTExMTExMTExMTExMTEifQ==",
+                optimizeProposition.getScope());
+        Assert.assertTrue(optimizeProposition.getScopeDetails().isEmpty());
+        Assert.assertEquals(1, optimizeProposition.getOffers().size());
+
+        final Offer offer = optimizeProposition.getOffers().get(0);
+        Assert.assertEquals("xcore:personalized-offer:1111111111111111", offer.getId());
+        Assert.assertEquals("10", offer.getEtag());
+        Assert.assertEquals(
+                "https://ns.adobe.com/experience/offer-management/content-component-html",
+                offer.getSchema());
+        Assert.assertEquals(OfferType.HTML, offer.getType());
+        Assert.assertEquals("<h1>This is a HTML content</h1>", offer.getContent());
+        Assert.assertEquals(1, offer.getCharacteristics().size());
+        Assert.assertEquals("true", offer.getCharacteristics().get("testing"));
+        Assert.assertNull(offer.getLanguage());
```

Comment:
can we also add assertion for preview cache here, as it will contain some proposition in this case.

Reply:
Added assertions for preview cache